src/audio_pixel.h

```diff
@@ -37,28 +40,64 @@ class audio_pixel_t {
 class audio_pixel_block_t {
 
 public: 
-    // TODO: handle the accessor not being valid
-    audio_pixel_block_t(MediaItem_Take* take, double pix_per_s) 
-        : m_take(take), m_pix_per_s(pix_per_s), 
-          m_accessor(safe_audio_accessor_t(take)) 
-          { /* TODO: */}
+    audio_pixel_block_t(double pix_per_s)
+        :m_pix_per_s(pix_per_s) {}
     ~audio_pixel_block_t() {}
 
-    void update() { /* TODO: see AudioAccessorStateChanged */}
+    void update(std::vector<double>& sample_buffer, int num_channels, int sample_rate) {
+        // constructs an audio pixel block using interleaved sample buffer and num channels
+        // clear the current set of audio pixels 
+        m_channel_pixels.clear();
+
+        // initalize the m_channel_pixels to have empty audio pixel vectors
+        for (int i = 0; i < num_channels; i++) {
+            std::vector<audio_pixel_t> empty_pixel_vec;
+            m_channel_pixels.push_back(empty_pixel_vec);
+        }
+
+        //calcualte the samples per audio pixel and initalize an audio pixel
+        int samples_per_pixel = m_pix_per_s * sample_rate;
+        audio_pixel_t curr_pixel = audio_pixel_t(0, -1, 0);
```

Comment:
why is max initialized to `0`, and min initialized to `-1`? what happens if the signal has a negative DC bias and the true signal max value is -1? 

to avoid stuff like this, we should have a default constructor for `audio_pixel_t` that initializes to extreme values, like `__DBL_MAX__` and `__DBL_MIN__`.